azure-mgmt-documentdb/src/main/java/com/microsoft/azure/management/documentdb/DatabaseAccount.java

```diff
@@ -243,45 +232,78 @@
      * Grouping of document db update stages.
      */
     interface Update extends
-        Resource.UpdateWithTags<Update>,
-        Appliable<DatabaseAccount>,
-        UpdateStages.WithConsistencyPolicy,
-        UpdateStages.WithIpRangeFilter {
+        UpdateStages.WithReadLocation,
+        UpdateStages.WithWriteLocations,
+        UpdateStages.WithOptionals {
     }
 
     /**
      * Grouping of document db update stages.
      */
     interface UpdateStages {
         /**
+         * Grouping of document db update stages.
+         */
+        interface WithOptionals extends
+            Resource.UpdateWithTags<WithOptionals>,
+            Appliable<DatabaseAccount>,
+            UpdateStages.WithConsistencyPolicy,
+            UpdateStages.WithIpRangeFilter {
+        }
+
+        /**
+         * The stage of the document db definition allowing the definition of a read location.
+         */
+        interface WithReadLocation {
+            /**
+             * A georeplication location for the DocumentDB account.
+             * @param region the region for the location
+             * @return the next stage
+             */
+            WithWriteLocations withReadableFailover(Region region);

```

Comment:
as discussed, there can only be one withWriteRegion and multiple ordered withReadRegions